src/game/main.cpp

```diff
@@ -162,6 +173,201 @@ void Check_Windowed(int argc, char *argv[])
+                    g_theWin32Mouse->Set_Ignore_Events(false);
+                }
+                break;
+            }
+            g_theMessageStream->Append_Message(GameMessage::MessageType::MSG_META_DEMO_INSTANT_QUIT);
+            return 0;
+
+        case WM_QUERYENDSESSION:
+            g_theMessageStream->Append_Message(GameMessage::MessageType::MSG_META_DEMO_INSTANT_QUIT);
+            return 0;
+
+        case WM_ERASEBKGND:
+            if (!g_gameNotFullscreen) {
+                return 1;
+            }
+            break;
+
+        case WM_ACTIVATEAPP: {
+            const bool activate_state = w_param == TRUE;
+            if (activate_state == g_gameActive) {
+                return 0;
+            }
+            g_gameActive = activate_state;
+            if (g_theGameEngine != nullptr) {
+                g_theGameEngine->Set_Is_Active(activate_state);
+            }
+            Reset_D3D_Device(activate_state);
+            if (activate_state && g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Cursor(g_theWin32Mouse->Get_Mouse_Cursor());
+            }
+            return 0;
+        }
+        case WM_NCHITTEST:
+            if (!g_gameIsWindowed) {
+                return HTCLIENT;
+            }
+            break;
+
+        case WM_KEYDOWN:
+            if (w_param == VK_ESCAPE) {
+                PostQuitMessage(EXIT_SUCCESS);
+            }
+            return 0;
+
+        case WM_SYSCOMMAND:
+            switch (w_param) {
+                case SC_MAXIMIZE:
+                case SC_SIZE:
+                case SC_MOVE:
+                case SC_KEYMENU:
+                case SC_MONITORPOWER:
+                    if (!g_gameIsWindowed) {
+                        return 0; // BUGFIX: vanilla returned 1 WinAPI says return 0
+                    }
+            }
+            break;
+
+        case WM_MOUSEMOVE: {
+            RECT window_rect{};
+            GetClientRect(g_applicationHWnd, &window_rect);
+            POINT mouse_position{ GET_X_LPARAM(l_param), GET_Y_LPARAM(l_param) };
+            if (mouse_position.x < window_rect.left || mouse_position.x > window_rect.right
+                || mouse_position.y < window_rect.top || mouse_position.y > window_rect.bottom) {
+                return 0;
+            }
+            if (g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Add_Win32_Event(WM_MOUSEMOVE, w_param, l_param, g_theMessageTime);
+            }
+            return 0;
+        }
+        case WM_MOUSEWHEEL: {
+            RECT window_rect{};
+            GetWindowRect(g_applicationHWnd, &window_rect);
+            POINT mouse_position{ GET_X_LPARAM(l_param), GET_Y_LPARAM(l_param) };
+            if (mouse_position.x < window_rect.left || mouse_position.x > window_rect.right
+                || mouse_position.y < window_rect.top || mouse_position.y > window_rect.bottom) {
+                return 0;
+            }
+        }
+            //[[fallthrough]]
```

Comment:
Instead of the fallthrough, how about we just copy up what it does below? This way we cut this entanglement.